oak_loader/src/options.rs

```diff
@@ -348,11 +348,22 @@ fn get_oidc_client_info(opt: &Opt) -> anyhow::Result<Option<ClientInfo>> {
 }
 
 /// Gets the default root TLS certificates from the embedded byte array.
-fn get_default_root_tls_certs() -> anyhow::Result<String> {
-    let result = std::str::from_utf8(include_bytes!("certs/roots.pem"))
-        .context("could not read embedded PEM-encoded root TLS certificates as a UTF8 string")?
-        .to_owned();
-    Ok(result)
+fn get_default_root_tls_certs() -> Vec<u8> {
+    include_bytes!("certs/roots.pem").to_vec()
+}
+
+/// Check that input byte vector represents a valid PEM formatted certificate.
+fn validate_certificate(certificate_bytes: &[u8]) -> anyhow::Result<()> {
+    let mut cursor = std::io::Cursor::new(certificate_bytes);
+
+    // We don't need the resulting certificate, but just check that a valid certificate can be
+    // created from the given byte array. `rustls` doesn't specify certificate parsing errors:
+    // https://docs.rs/rustls/0.17.0/rustls/internal/pemfile/fn.certs.html
+    rustls::internal::pemfile::certs(&mut cursor).map_err(|_err| {
```

Comment:
You can capture what you just said in words via an irrefutable pattern, which also shows the type, and will fail to match if the type of the error changes in the future:

```suggestion
    rustls::internal::pemfile::certs(&mut cursor).map_err(|()| {
```

Reply:
Of course! Thanks :)